sdk/python/generator/StructTypeFormatter.py

```diff
@@ -136,12 +136,13 @@ def generate_condition(self, field, prefix_field=False):
 		condition_operator = condition_to_operator_map[conditional.operation]
 
 		value = f'{conditional.value}'
-		yoda_value = value if condition_field.get_type().is_int else f'{condition_field.get_type().typename}.{value}'
+		condition_ast_model = condition_field.type_ast_model
+		yoda_value = value if DisplayType.INTEGER == condition_ast_model.display_type else f'{condition_ast_model.name}.{value}'
 		field_prefix = 'self.' if prefix_field else ''
 
 		# HACK: instead of handling dumb magic value in namespace parent_name, generate slightly simpler condition
-		if prefix_field and isinstance(field.ast_model.field_type, Array):
-			return f'if {field_prefix}{field.original_field_name}:\n'
+		if prefix_field and DisplayType.UNSET != field.ast_model.display_type:
```

Comment:
this looks bit too wide? (I assume it works correctly though)

Reply:
possibly, but it works so would leave it for now